crates/algorithms/src/haversine_formula/mod.rs

```diff
@@ -0,0 +1,81 @@
+use crate::PI;
+use nalgebra::{Point2};
+///function to convert degree to radians
+/// we use this func because coordinate is in degree, and we need to use trigonometric function
+fn to_radians(degrees: f64) -> f64 {
+    degrees * PI / 180.0
+}
+/// function to convert radians to degree
+fn to_degrees(radians: f64) -> f64 {
+    radians * 180.0 / PI
+}
+///calculate distance between point with use haversine formula
+/// return distance in km
+fn return_distance_between_points(p1: Point2<f64>, p2: Point2<f64>) -> f64 {
+    //we need radius of earth
+    let earth_radius_km = 6371.0;
+    //extract coordinate from point
+    let lat1 = p1.x;
```

Comment:
If lat1 and the rest are never used in any other context, why not use 
let lat1 = p1.x.to_degrees() and be done with it?